File: frontend/src/types/order.ts
export type OrderStatus = "CREATED" | "ACCEPTED" | "SHIPPED" | "COMPLETED" | "CANCELED";

export type OrderItem = {
  dishId: string;
  dishName: string;
  dishPrice: number; // ✅ 正确字段名
  quantity: number;
  restaurantId: string;
  subtotal?: number;
};

export type Order = {
  id: string;
  createdAt: string;
  status: OrderStatus; // ✅ 用联合类型限制 status
  recipient: string;
  phone: string;
  address: string;
  totalPrice: number;
  restaurantName: string;
  items: OrderItem[];
};

File: backend/order-service/src/main/java/com/shaluo/orderservice/model/postgres/Order.java
package com.shaluo.orderservice.model.postgres;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Data
@Entity
@Table(name = "orders")
public class Order {

    @Id
    @Column(columnDefinition = "uuid", updatable = false)
    private UUID id;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    private String address;

    private String recipient;

    private String phone;

    @Column(nullable = false)
    private String restaurantId;

    @Column(nullable = true)
    private String restaurantName;

    private LocalDateTime createdAt;


    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OrderStatus status;

    @Column(nullable = false)
    private Double totalPrice;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    private List<OrderItem> items;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}
